### PR TITLE
use make_X API follow-up

### PR DESCRIPTION
**Pull request by kroening**
This is follow up from #4177.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: src/goto-instrument/dump_c.cpp [hunk 1]
```diff
@@ -593,7 +593,7 @@ void dump_ct::cleanup_decl(
   tmp.add(goto_programt::make_decl(decl.symbol()));
 
   if(value.is_not_nil())
-    tmp.add(goto_programt::make_assignment(code_assignt(decl.op0(), value)));
+    tmp.add(goto_programt::make_assignment(decl.op0(), value));
```
Comment (tautschnig) [inline src/goto-instrument/dump_c.cpp]: While updating this line, could we also use `decl.symbol()` instead of `decl.op0()`

**Review by tautschnig — APPROVED** [review]

File: src/goto-instrument/dump_c.cpp (hunk 1 above)
Comment (kroening) [reply src/goto-instrument/dump_c.cpp]: done

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: 2203dcddc748291f0dd594bf5cc5e9cfbab8ad69).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/100902365
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: 610d8367b7a81e54f579555b271c19b67dcf62f5).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/100907765
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 2631fcbc9b4121acd4333ebc0d451edaaaf85caf).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/100910867

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: a5deeb0f50f3f0d5805666865f2c7a289461daf5).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/101183449

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 5520e058e8ac088f2497cb61296ec6621e0bc918).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/101188152

**Review by thk123 — APPROVED** [review]

**Review by chrisr-diffblue — APPROVED** [review]

**Review by tautschnig — CHANGES_REQUESTED** [review]
(no action by anyone else needed, I'll take care of it) #4278 needs to be merged before this one to make sure the bug and bug fix are documented. I will then take care of rebasing this one on top of it.

**Review by tautschnig — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
:warning:
This PR failed Diffblue compatibility checks (cbmc commit: cba52e3a399347b045e8badb04d531296ef7d504).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/102331367
Status will be re-evaluated on next push.
Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

  - the compatibility was already broken by an earlier merge.

**Pull request merged**